core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -96,7 +128,56 @@ class DefaultAutoTopicCreationManager(
     requestContext: RequestContext
   ): Unit = {
     if (topics.nonEmpty) {
-      sendCreateTopicRequest(topics, Some(requestContext))
+      sendCreateTopicRequestWithErrorCaching(topics, Some(requestContext))
+    }
+  }
+
+  override def getTopicCreationErrors(
+    topicNames: Set[String],
+    errorCacheTtlMs: Long
```

Comment:
Passing the TTL here is incorrect. The TTL is defined group-specific, so you are expiring topic creation errors for one group with the TTL for a different group. 

I think we need to pass the errorCacheTTL into `createStreamsInternalTopics`, and store the expiry time instead of the error receival timestamp in `CachedTopicCreationError`.